Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -12,28 +12,32 @@ public enum ActorStates
 
 public class ActorPhysics : ActorComponent
 {
+	private new Transform transform;
```

Comment:
Yeah, I've noticed that when doing things like gizmos and editor scripting it causes problems. Unity also has `Reset()` which get's called in the editor, but if that doesn't work for some reason we could do lazy initialization like we're doing `instance` for `Singleton`.